### ISSUE TITLE
better defaults for magnetic sld (Trac #1004)

### ISSUE THREAD

**Issue by pkienzle** [SasView/sasview#1073]
When turning on magnetism, the default magnetic sld is set to 1e-6.  This should default to 1 instead.

When turning off magetism and turning it on again, the defaults are reset.

'''NOTE'''
The rest of this ticket related to failure to compute is moved to #1134

Migrated from http://trac.sasview.org/ticket/1004
```json
{
    "status": "closed",
    "changetime": "2018-11-16T00:46:22",
    "_ts": "2018-11-16 00:46:22.534298+00:00",
    "description": "When turning on magnetism, the default magnetic sld is set to 1e-6.  This should default to 1 instead.\n\nWhen turning off magetism and turning it on again, the defaults are reset.\n\n'''NOTE'''\nThe rest of this ticket related to failure to compute is moved to #1083",
    "reporter": "pkienzle",
    "cc": "",
    "resolution": "fixed",
    "workpackage": "SasView Bug Fixing",
    "time": "2017-09-29T16:42:51",
    "component": "SasView",
    "summary": "better defaults for magnetic sld",
    "priority": "minor",
    "keywords": "",
    "milestone": "SasView 4.2.0",
    "owner": "butler",
    "type": "defect"
}
```

**Comment by butlerpd** [comment]
Trac update at `2018/02/13 14:33:52`: 
* **butler** changed owner from "" to "butler"
* **butler** changed status from "new" to "assigned"

**Comment by butlerpd** [comment]
Trac update at `2018/02/13 14:34:26`: **butler** changed priority from "major" to "critical"

**Comment by ajj** [comment]
Trac update at `2018/03/20 14:27:00`: **ajj** changed priority from "critical" to "blocker"

**Comment by butlerpd** [comment]
Trac update at `2018/04/02 23:34:08`: 
* **butler** changed description from:

> When turning on magnetism, the default magnetic sld is set to 1e-6.  This should default to 1 instead.
> 
> When turning off magetism and turning it on again, the defaults are reset.
> 
> Also, 4.1.2 does not appear to compute magnetic models. Try for example, a sphere with:
> 
> ```
> sld: 4.50078  M0:0.317  mtheta:19.8  mphi:154.5
> sld_solvent: 3.75713  M0:4.385  mtheta:-51.4  mphi:155.1
> radius: 200
> ```
> and compare with the equivalent from sascomp:
> ```
> $ sascomp sphere -magnetic -nq=400 -pars -random=547121 radius=200
> ```
> 
> 

to:

> When turning on magnetism, the default magnetic sld is set to 1e-6.  This should default to 1 instead.
> 
> When turning off magetism and turning it on again, the defaults are reset.
> 
> '''NOTE'''
> The rest of this ticket related to failure to compute is moved to #1083


* **butler** changed priority from "blocker" to "minor"

**Comment by butlerpd** [comment]
Trac update at `2018/06/15 14:10:09`: 
* **butler** commented:

probably not going to make it for this release so moving to 4.3


* **butler** changed milestone from "SasView 4.2.0" to "SasView 4.3.0"

**Comment by pkienzle** [comment]
Trac update at `2018/10/03 18:39:20`: **pkienzle** commented:

There is a PR for addressing the order of magnitude portion of the ticket.

To address the main point of the ticket (that the values return the defaults when magnetism is toggled off and on), need to add a flag to the sasmodels wrapper that determines whether magnetism should be enabled for the calculation, and ignore the magnetic parameters otherwise.  Then don't modify `_on_mag_on()` method in basepage.py within sasview so that it sets/resets this flag rather than setting/resetting the magnetic slds all to zero.

**Comment by pkienzle** [comment]
Trac update at `2018/11/15 15:48:54`: 
* **pkienzle** commented:

Resolved in 4.2 on 2018-10-04


* **pkienzle** changed resolution from "" to "fixed"
* **pkienzle** changed status from "assigned" to "closed"

**Comment by butlerpd** [comment]
Trac update at `2018/11/16 00:46:22`: **butler** changed milestone from "SasView 4.3.0" to "SasView 4.2.0"
